File: source/algorithm/NQ077/NQ077_improved.py
def main():
    import sys
    q = int(sys.stdin.readline().strip())
    while q > 0:
        diffs = []
        line = sys.stdin.readline().strip()
        n, k = map(int, line.split())
        line = sys.stdin.readline().strip()
        a = list(map(int, line.split()))
        for i in range(n-1):
            for j in range(i+1, n):
                diff = abs(a[i] - a[j])
                diffs.append(diff)  # 计算所有可能的差
        diffs.sort(reverse=True)  # 降序排序
        unique_diffs = []
        seen = set()
        for d in diffs:
            if d not in seen:
                seen.add(d)
                unique_diffs.append(d)  # 去重
        if k <= len(unique_diffs):
            print(unique_diffs[k-1])  # 输出第K大的差
        else:
            print("Invalid k value!")
        q -= 1

if __name__ == "__main__":
    main()

File: source/algorithm/NQ077/NQ077_improved.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;
const int MAXN = 1007;
int a[MAXN];
int main() {
    int q;
    cin >> q;
    while (q--) {
        vector<int> diffs;
        int n, k;
        cin >> n >> k;
        for (int i = 0; i < n; ++i) {
            cin >> a[i];
        }
        for (int i = 0; i < n-1; ++i) {
            for (int j = i+1; j < n; ++j) {
                int diff = abs(a[i] - a[j]);
                diffs.push_back(diff);  // 计算所有可能的差
            }
        }
        sort(diffs.begin(), diffs.end(), greater<int>());  // 降序排序
        auto last = unique(diffs.begin(), diffs.end());
        diffs.erase(last, diffs.end());  // 去重
        if (k <= diffs.size()) {
            cout << diffs[k-1] << endl;  // 输出第K大的差
        } else {
            cout << "Invalid k value!" << endl;
        }
    }
    return 0;
}